react-components/src/components/RevealContainer/RevealResourcesContext.tsx

```diff
@@ -0,0 +1,38 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+
+import { type CogniteModel } from '@cognite/reveal';
+import { type ReactElement, createContext, useContext, useState, type ReactNode } from 'react';
+
+type RevealResourcesContextType = {
+  reveal3DResources: CogniteModel[];
+  update3DResources: (newResources: CogniteModel[]) => void;
```

Comment:
As mentioned in other comments I think it should use the same setState function that is returned from useState.